paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -694,11 +692,23 @@ public static Catalog createHiveCatalog(CatalogContext context) {
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
+
+        /** Hive catalog only support hive lock. */
+        if (options.getOptional(LOCK_ENABLED).orElse(false)) {
+            Optional<String> lockType = options.getOptional(LOCK_TYPE);
+            if (lockType.isPresent()) {
+                checkArgument(
+                        LOCK_IDENTIFIER.equals(lockType.get()),
```

Comment:
We have abstracted `lockFactory` into `AbstractCatalog`. Do we still need such restrictions?

Reply:
At the same time, we can delete the `lockFactory` in `HiveCatalog`.

Reply:
I think we can set `hive` for `LOCK_TYPE` if there's no lock type in hive catalog, I will update this PR